src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -1177,63 +1177,44 @@ namespace cryptonote
-          << o.target.type().name() << ", expected txout_to_carrot_v1 in transaction id=" << get_transaction_hash(tx));
-      }
-      else if (hf_version == HF_VERSION_CARROT)
-      {
-        // during v17, require outputs be of type txout_to_tagged_key OR txout_to_carrot_v1
-        // to allow grace period before requiring all to be txout_to_carrot_v1
-        const std::type_info &o_type = o.target.type();
-        CHECK_AND_ASSERT_MES(
-          o_type == typeid(txout_to_carrot_v1) || o_type == typeid(txout_to_tagged_key),
-          false, "wrong variant type: " << o_type.name()
-          << ", expected txout_to_carrot_v1 or txout_to_tagged_key in transaction id=" << get_transaction_hash(tx));
-
-        // require all outputs in a tx be of the same type
-        const std::type_info &first_type = tx.vout.at(0).target.type();
-        CHECK_AND_ASSERT_MES(o_type == first_type, false, "non-matching variant types: "
-          << o_type.name() << " and " << first_type.name() << ", "
-          << "expected matching variant types in transaction id=" << get_transaction_hash(tx));
-      }
-      else if (hf_version > HF_VERSION_VIEW_TAGS)
-      {
-        // from v16, require outputs have view tags
-        CHECK_AND_ASSERT_MES(o.target.type() == typeid(txout_to_tagged_key), false, "wrong variant type: "
-          << o.target.type().name() << ", expected txout_to_tagged_key in transaction id=" << get_transaction_hash(tx));
-      }
-      else if (hf_version < HF_VERSION_VIEW_TAGS)
-      {
-        // require outputs to be of type txout_to_key
-        CHECK_AND_ASSERT_MES(o.target.type() == typeid(txout_to_key), false, "wrong variant type: "
-          << o.target.type().name() << ", expected txout_to_key in transaction id=" << get_transaction_hash(tx));
-      }
-      else  //(hf_version == HF_VERSION_VIEW_TAGS)
-      {
-        // require outputs be of type txout_to_key OR txout_to_tagged_key
-        // to allow grace period before requiring all to be txout_to_tagged_key
-        CHECK_AND_ASSERT_MES(o.target.type() == typeid(txout_to_key) || o.target.type() == typeid(txout_to_tagged_key), false, "wrong variant type: "
-          << o.target.type().name() << ", expected txout_to_key or txout_to_tagged_key in transaction id=" << get_transaction_hash(tx));
-
-        // require all outputs in a tx be of the same type
-        CHECK_AND_ASSERT_MES(o.target.type() == tx.vout[0].target.type(), false, "non-matching variant types: "
-          << o.target.type().name() << " and " << tx.vout[0].target.type().name() << ", "
-          << "expected matching variant types in transaction id=" << get_transaction_hash(tx));
-      }
+      const std::type_info &cur_type = tx.vout.at(0).target.type();
+      CHECK_AND_ASSERT_MES(cur_type == o_type, false, "non-matching variant types: "
+        << o_type.name() << " and " << cur_type.name() << ", "
+        << "expected matching variant types in transaction id=" << get_transaction_hash(tx));
```

Comment:
```suggestion
      const std::type_info &cur_type = o.target.type();
      CHECK_AND_ASSERT_MES(cur_type == o_type, false, "non-matching variant types: "
        << o_type.name() << " and " << cur_type.name() << ", "
        << "expected matching variant types in transaction id=" << get_transaction_hash(tx));
```

Reply:
Oops that was a pretty big mistake..... thanks for catching that